tests/cases/layer.slint

```diff
@@ -0,0 +1,61 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+import { Button } from "std-widgets.slint";
+
+// Test case for manual visual verification that layering does not pessimize the
+// output, and to have a compile-time verification that the lowering to the Layer
+// element compiles.
+
+export TestCase := Window {
+    preferred-width: 800px;
+    preferred-height: 600px;
+    background: white;
+
+    VerticalLayout {
+
+        Button {
+            text: "Press me to start the animation and check that it is smooth.";
+            clicked => {
+                layered.place_to_the_right = true;
+            }
+        }
+
+        Rectangle {
+
+            // This will be a layer
+            layered := Rectangle {
+                cache-rendering-hint: true;
+                width: 200px;
+                height: 100px;
+
+                property <bool> place_to_the_right;
+                states [
+                    right when place_to_the_right: {
+                        x: 200px;
+                    }
+                    left when !place_to_the_right: {
+                        x: 10px;
+                    }
+                ]
+
+                animate x {
+                    duration: 15s;
+                }
+
+                background: red;
+                
+                for i in 1000: Rectangle {
```

Comment:
```suggestion
                for i in 1000: Rectangle {
                    cache-rendering-hint: i == 8;
```

Just some silly test with a condition and in a repeater

Reply:
Good idea